sdk/python/feast/data_source.py

```diff
@@ -767,6 +774,17 @@ def get_table_column_names_and_types(self) -> Iterable[Tuple[str, str]]:
 
         return name_type_pairs
 
+    def _table_exists(self) -> bool:
+        from google.api_core.exceptions import NotFound
+        from google.cloud import bigquery
+
+        client = bigquery.Client()
```

Comment:
Should this client receive configuration?

Reply:
Do you mean like the following?
```python
gcp_project, _ = bq_table_ref.split(":")
bq_client = bigquery.Client(project=gcp_project)
```

I saw in line https://github.com/feast-dev/feast/blob/master/sdk/python/feast/data_source.py#L745 and a lot of other instances where it simply just has `client = bigquery.Client()` without specifying the project, do you know when and when it's not necessary?

Reply:
i think calling Client w/o specifying project should be fine. https://googleapis.dev/python/bigquery/latest/_modules/google/cloud/bigquery/client.html#Client.get_table says get_tables uses from_string https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.table.TableReference.html#google.cloud.bigquery.table.TableReference.from_string which parses the string for the project and dataset if a default project id isn’t specified